renku/core/login.py

```diff
@@ -0,0 +1,283 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2022 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Logging in to a Renku deployment."""
+
+import os
+import time
+import urllib
+import webbrowser
+from typing import TYPE_CHECKING, Optional, cast
+
+from pydantic import validate_arguments
+
+from renku.core import errors
+from renku.core.config import get_value, remove_value, set_value
+from renku.core.util import communication
+from renku.core.util.git import RENKU_BACKUP_PREFIX, create_backup_remote, get_remote, get_renku_repo_url
+from renku.core.util.urls import parse_authentication_endpoint
+from renku.domain_model.enums import ConfigFilter
+from renku.domain_model.project_context import project_context
+
+if TYPE_CHECKING:
+    from renku.infrastructure.repository import Repository
+
+
+CONFIG_SECTION = "http"
+KEYCLOAK_REALM = "Renku"
+CLIENT_ID = "renku-cli"
+
+
+@validate_arguments(config=dict(arbitrary_types_allowed=True))
+def login(endpoint: Optional[str], git_login: bool, yes: bool):
+    """Log into a Renku deployment."""
+    from renku.core.util import requests
+
+    try:
+        repository = project_context.repository
+    except ValueError:
+        repository = None
+
+    parsed_endpoint = _parse_endpoint(endpoint)
+
+    remote_name, remote_url = None, None
+    if git_login:
+        if repository is None:
+            communication.warn("Cannot log in to git outside a project.")
```

Comment:
Should this mention `--no-git` ?

Reply:
I don't think so. This is just a warning and won't hurt if they don't pass it. `--no-git` is mentioned in the docs.